src/webgpu/api/validation/utils.ts

```diff
@@ -0,0 +1,425 @@
+interface Resource {
+  buffer?: GPUBufferBindingLayout;
+  sampler?: GPUSamplerBindingLayout;
+  texture?: GPUTextureBindingLayout;
+  storageTexture?: GPUStorageTextureBindingLayout;
+  externalTexture?: GPUExternalTextureBindingLayout;
+  code: string;
+  staticUse?: string;
+}
+
+export const kAPIResources: Resource[] = [
```

Comment:
nit: The fields and the array should be readonly, to avoid tests accidentally modifying global stuff

```suggestion
interface Resource {
  readonly buffer?: GPUBufferBindingLayout;
  readonly sampler?: GPUSamplerBindingLayout;
  readonly texture?: GPUTextureBindingLayout;
  readonly storageTexture?: GPUStorageTextureBindingLayout;
  readonly externalTexture?: GPUExternalTextureBindingLayout;
  readonly code: string;
  readonly staticUse?: string;
}

export const kAPIResources: readonly Resource[] = [
```